crates/abq_hosted/src/config.rs

```diff
@@ -285,21 +317,58 @@ mod test {
             ))
             .create();
 
-        let HostedQueueConfig {
-            addr,
-            run_id,
-            auth_token,
-            tls_public_certificate,
-            rwx_access_token_kind,
-        } = HostedQueueConfig::from_api(server.url(), &test_access_token(), &in_run_id)
-            .await
-            .unwrap();
+        if let HostedQueueConfig::Success(config) =
+            HostedQueueConfig::from_api(server.url(), &test_access_token(), &in_run_id)
+                .await
+                .unwrap()
+        {
+            assert_eq!(config.addr, "168.220.85.45:8080".parse().unwrap());
+            assert_eq!(config.run_id, in_run_id);
+            assert_eq!(config.auth_token, test_auth_token());
+            assert!(config.tls_public_certificate.is_none());
+            assert_eq!(config.rwx_access_token_kind, AccessTokenKind::Personal)
+        } else {
+            unreachable!("expected success config")
+        }
+    }
+
+    #[tokio::test]
+    async fn get_hosted_queue_config_unsupported_usage() {
+        let mut server = Server::new();
+
+        let in_run_id = RunId("1234".to_string());
 
-        assert_eq!(addr, "168.220.85.45:8080".parse().unwrap());
-        assert_eq!(run_id, in_run_id);
-        assert_eq!(auth_token, test_auth_token());
-        assert!(tls_public_certificate.is_none());
-        assert_eq!(rwx_access_token_kind, AccessTokenKind::Personal)
+        let _m = server
+            .mock("GET", "/queue")
+            .match_header(
+                "Authorization",
+                format!("Bearer {}", test_access_token()).as_str(),
+            )
+            .match_header("User-Agent", format!("abq/{}", abq_utils::VERSION).as_str())
+            .match_query(Matcher::AnyOf(vec![Matcher::UrlEncoded(
+                "run_id".to_string(),
+                in_run_id.to_string(),
+            )]))
+            .with_status(200)
+            .with_header("content-type", "application/json")
+            .with_body(
+                r#"{{"usage_error":"This usage is not supported","rwx_access_token_kind":"personal_access_token"}}"#,
+            )
+            .create();
+
+        if let HostedQueueConfig::Unsupported(config) =
+            HostedQueueConfig::from_api(server.url(), &test_access_token(), &in_run_id)
+                .await
+                .unwrap()
```

Comment:
For some reason this seems to be unable to parse the response successfully. I'm pretty sure this is a problem with my setup as the integration test seems to work fine.

Spot anything :eyes:?

Reply:
My best guess is the integration test does this:

```
                json!({
                    "usage_error": "Cannot use personal access token with run ID that doesn't exist",
                    "rwx_access_token_kind": "personal_access_token",
                }).to_string()
```

But I'm not seeing why the other tests in this file wouldn't also have this issue 🤔

Reply:
Yep, that was it, but I still don't quite understand why 😅